Documentation/StringFormatting.md

```diff
@@ -0,0 +1,128 @@
+
+For basic usage, any occurrences of `{}` in the format string are replaced with the other arguments, converted to string form, in order:
+
+```c++
+String::formatted("Well, {} my {} friends!", "hello", 42) == "Well, hello my 42 friends!";
+```
+
+If you want to include a literal `{` in the output, use `{{`:
+
+```c++
+String::formatted("{{ {}", "hello") == "{ hello";
+```
+
+You can refer to the arguments by index, if you want to repeat one or change the order:
+```c++
+String::formatted("{3}{0}{1}", "a", "b", "c", "d") == "dab";
+```
+
+To control how the arguments are formatted, add colon after the optional index, and then add format specifier characters:
+
+```c++
+String::formatted("{:.4}", "cool dude") == "cool";
+String::formatted("{0:.4}", "cool dude") == "cool";
+```
+
+## Format specifiers
+
+In order, the format can contain:
+- Fill character and alignment
+- Sign
+- `#` Hash
+- `0` Zero
+- Width
+- Precision
+- Type
+
+Each of these is optional. You can include any combination of them, but they must be in this order.
+
+### Fill and alignment
+
+This is an optional fill character, followed by an alignment. The fill character can be anything apart from `{` or `}`, and is used to fill any space left when the input has fewer characters than the format requests. By default, it is a space. (` `)
+
+The alignment characters are:
+- `<`: Align left.
+- `>`: Align right.
+- `^`: Align centered.
+
+### Sign
+
+- `+`: Always display a sign before the number.
+- `-`: Display a sign for negative numbers only.
+- (space): Display a sign for negative numbers, and a leading space for other numbers.
+
+### Hash
+
+`#` causes an "alternate form" to be used.
+
+For integer types, this adds the number-base prefix after the sign:
+- `0b` for binary.
+- `0` for octal.
+- `0x` for hexadecimal.
+
+### Zero
+
+`0` pads the number with leading zeros.  
+
+### Width and Precision
+
+The width defines the minimum number of characters in the output. The precision is a `.` followed by a precision number, which is used as the precision of floating-point numbers, or a maximum-width for string values.
+
+Both the width and precision can be provided as a replacement field (`{}`, optionally including an argument index) which allows you to use an integer argument instead of a hard-coded number.
+
+### Type
+
+| Type     | Effect                | Example output           |
+|----------|-----------------------|--------------------------|
+| b        | binary                | `110`, `0b000110`        |
+| B        | binary uppercase      | `110`, `0B000110`        |
+| d        | decimal               | `42`, `+0000042`         |
+| o        | octal                 | `043`                    |
+| x        | hexadecimal           | `ff0`, `0x00000ff0`      |
+| X        | hexadecimal uppercase | `FF0`, `0X00000FF0`      |
+| c        | character             | `a`                      |
+| s        | string (default)      | `well, hello friends!`   |
```

Comment:
Dunno if it makes sense to say string is the default type. The default depends on the type of the value being formatted, no? e.g. integral values default to decimal.

Reply:
I was making an assumption there, I see now that there's a distinct `Default` formatter mode apart from String, so I've updated the table accordingly.